virttest/qemu_devices/qbuses.py

```diff
@@ -650,6 +650,118 @@ def _increment_addr(self, addr, last_addr=None):
         return super(QPCIBus, self)._increment_addr(addr, last_addr=last_addr)
 
 
+class QPCIEBus(QPCIBus):
+    """
+    PCIE root bus representation (creates pcie-root-port for virtio pci
+    device by default.)
+    """
+
+    def __init__(self, busid, bus_type, aobject=None,
+                 root_type="pcie-root-port"):
+        """
+        :param busid: id of the bus (pcie.0)
+        :param bus_type: type of the bus (PCIE)
+        :param aobject: Related autotest object (pci.0)
+        :param root_type: root port type
+        """
+        super(QPCIEBus, self).__init__(busid, bus_type, aobject)
+        self.__root_ports = {}
+        self.__root_type = root_type
+
+    def is_direct_plug(self, device):
+        """
+        Justify if the device should be plug to pcie bus directly,
+        the device is not directly plugged only if it is a virtio
+        pci device and the device param direct_plug is "no".
+
+        :param device: the QBaseDevice object
+        :return: bool value for directly plug or not.
+        """
+        pcie_devices = ["virtio-net-pci", "virtio-blk-pci",
+                        "virtio-scsi-pci", "virtio-balloon-pci",
+                        "virtio-serial-pci", "virtio-rng-pci",
+                        "e1000e", "virtio-gpu-device"]
```

Comment:
`nec-xhci-pci` should be in list as well?

Reply:
@luckyh I am not sure, but I think it should be plug to pcie.0 directly, @jingzhao84 could you help confirm?

Reply:
@luckyh nec-xhci-pci can be plug into pcie-root-port and pcie.0, but the best topology is that plug into pcie.0 through the develop's recommend, so I think we can set it in cfg file if we want to plug into pcie-root-port.
Thanks